Description update for Redis installation on debian based systems

mmattel opened the pull request:
Update to better describe the basic installation of redis on debian based systems.

mmattel commented:
This completes the description for properly installing Redis when you use Ubuntu 14.04.
Please also see the PR #2000 where this was originally discussed.

mmattel commented:
Refernce issue: #2043 (Ubuntu 14.04 - steps to install working php redis module)

RealRancor commented on admin_manual/configuration_server/caching_configuration.rst:
```diff
@@ -156,19 +156,66 @@ that cached objects are available for as long as they are needed.
+  
+  One solution is to look for a PPA. See `launchpad <https://launchpad.net/ubuntu/+source/php-redis>`_.
+  
+  Another and preferred solution option is using the PECL/PEAR PHP extension repository:
```
I don't think this should be the preferred solution as you won't get updates if installed manually. I also don't know if this can cause troubles if the system is upgraded to a newer release like 15.04/15.10/16.04

RealRancor commented on admin_manual/configuration_server/caching_configuration.rst:
```diff
@@ -156,19 +156,66 @@ that cached objects are available for as long as they are needed.
  ps ax | grep redis
  22203 ? Ssl    0:00 /usr/bin/redis-server 127.0.0.1:6379 
  
+| You may also check the presense of the Redis configuration links in the corresponding subdirectories of php ``cli`` and ``fpm`` (if you use these options).
```
I don't think you need redis for cli in oC 8.2+

RealRancor commented:
Besides my comments :+1:

mmattel commented:
I have rewritten the redis part:
- Split into Server and Instance blocks
- Itemized Operating Systems
- General setup section
- Notes
- ect...

@RealRancor imho, occ is php-cli and therefore adding the redis link to the cli component of the php setup is correct.

RealRancor commented:
> imho, occ is php-cli and therefore adding the redis link to the cli component of the php setup is correct.

IMHO php-cli shouldn't use caching. This here is for APCu but probably also valid for redis caching:

https://stackoverflow.com/questions/1245242/php-apc-in-cli-mode

mmattel commented:
Do the following:
- open three shell´s
- 1st shell: redis-cli monitor
- 2nd shell: check for precense of ``20-redis.ini -> ../../mods-available/redis.ini`` 
--> should be present
- 3rd shell: sudo -uwww-data ./occ

Now just watch that redis is doing a lot...
- 2nd shell: unlink ``20-redis.ini``
- 3rd shell: sudo -uwww-data ./occ

Now just watch that redis is doing nothing...

RealRancor commented:
@mmattel If it is doing something doesn't mean it is useful that it is done. ;-)

mmattel commented:
I did following test:

**Without having** ``cli/conf.d/20-redis.ini``
```
sudo -uwww-data ./occ files:scan --all
Starting scan for user 1 out of 1 (xxxx)

+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 3474    | 28274 | 00:01:39     |
+---------+-------+--------------+
```
**Having** ``cli/conf.d/20-redis.ini``
```
sudo -uwww-data ./occ files:scan --all
Starting scan for user 1 out of 1 (xxxx)

+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 3474    | 28274 | 00:01:18     |
+---------+-------+--------------+
```

Can we agree that it is positive for ``cli`` to have the redis link ?

phil-davis commented on admin_manual/configuration_server/caching_configuration.rst:
```diff
@@ -154,48 +154,141 @@ as a local cache for :doc:`Transactional File Locking
+  
+  See also: https://pecl.php.net/package/redis
+
+| **Server Preperation**
```
s/Preperation/Preparation/

RealRancor commented:
@mmattel I think one single test is really not enough to prove anything. But go for it if you think it should be in there.

Personally i just want to avoid additional questions / support effort caused by users not familiar with that and doing stuff like breaking their system by modifying additional files.

phil-davis commented on admin_manual/configuration_server/caching_configuration.rst:
```diff
@@ -154,48 +154,141 @@ as a local cache for :doc:`Transactional File Locking
-For best performance, use Redis for file locking by adding this::
-
-  'memcache.locking' => '\OC\Memcache\Redis',
+| You may also check the presense of the PHP Redis configuration links in the corresponding subdirectories of php ``cli`` and ``fpm``. If you use nginx, you most likely also use fpm.
```
s/presense/presence/

RealRancor commented:
Ok, it *could* speedup things if the webserver is also using redis as a memcache and both are sharing / using the same stuff:

https://stackoverflow.com/questions/3722595/how-to-share-apc-cache-between-apache-and-php-cli

mmattel commented:
@phil-davis I missed you :smile:, just updated, thanks

phil-davis commented:
@mmattel I have been busy doing lots of stuff for the upcoming https://github.com/pfsense/pfsense 2.3 - I hope to get back to more ownCloud soon and can hopefully contribute code fixes as well as mindless documentation grammar and typo stuff :)

mmattel commented:
@carlaschroder corrected the subheadings as suggested. The startup of redis is described in the text (automatic startup on Debian based systems). Startup has not changed to what it was before...

carlaschroder commented:
Whoa everyone, just stop please :)  @mmattel, you have worked hard on this and I thank you.
Now here are the issues I have to consider:
-- Maintaining this documentation
-- Which has veered way over the line into teaching system administration, which is something I try to avoid. Especially when it comes to manually configuring .ini files, which leads to madness and tooth decay
-- Yet we are stuck with having to do this, because ownCloud only supports the second most recent Redis release (why?)
-- Many Linux distros do not package the 2.2.5 Redis release, and especially the enterprise distros which are always conservative
-- Using PECL risks getting out of sync with package managers, which creates a new set of problems

So this is what I am going to do. I am going to put the PECL sections in the documentation Wiki. The official manuals will include instructions for installing the correct Redis version with distro package managers using third-party repos. Thanks again for all the work you put into this; splitting it between the official manuals and the Wiki makes it manageable, and all I have to do is copy-paste.

carlaschroder commented:
Closing for  https://github.com/owncloud/documentation/pull/2065. Thanks everyone!
